cloud/ec2.go

```diff
@@ -1203,6 +1203,11 @@ func (m *ec2Manager) StartInstance(ctx context.Context, h *host.Host, user strin
 	}
 
 	var instance *ec2.Instance
+
+	if err = h.SetIPv4Address(*instance.PublicIpAddress); err != nil {
```

Comment:
We can't do this here because we haven't set instance to anything yet. But actually we call cacheHostData at line 1232 which should already handle setting the ipv4 address.

Reply:
removed setIpv4 function and combined with cacheHostData